src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -194,7 +203,57 @@ class ContentOpenWith extends PureComponent<Props, State> {
      * @return {void}
      */
     fetchOpenWithSuccessHandler = (integrations: Array<Integration>): void => {
-        this.setState({ integrations, isLoading: false });
+        const boxEditIntegration = integrations.find(integration => {
+            return this.isBoxEditIntegration(integration);
+        });
+
+        if (boxEditIntegration && !boxEditIntegration.isDisabled) {
+            this.boxEdit = new BoxEdit();
+            // Check if Box Tools is installed and reachable
+            this.boxEdit
+                .checkBoxEditAvailability()
+                .catch(() => {
+                    boxEditIntegration.isDisabled = true;
+                    boxEditIntegration.disabledReasons = ['Install Box Tools to open this file on your desktop'];
+                })
+                .then(this.getFileExtension)
+                .then(extension => {
```

Comment:
Can this be migrated to another top level function, so that >>> arrowness reduces?